python/rmm/_lib/logger.pyx

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+cdef extern from "spdlog/common.h" namespace "spdlog::level" nogil:
+    cdef enum level_enum:
+        trace = 0
+        debug = 1
+        info = 2
+        warn = 3
+        err = 4
+        critical = 5
+        off = 6
+
+
+cdef extern from "spdlog/spdlog.h" namespace "spdlog" nogil:
+    cdef cppclass spdlog_logger "spdlog::logger":
+        spdlog_logger() except +
+        void set_level(level_enum log_level) except +
+        level_enum level() except +
+        void flush() except +
+        void flush_on(level_enum log_level) except +
+        level_enum flush_level() except +
+
+
+cdef extern from "rmm/logger.hpp" namespace "rmm" nogil:
+    cdef spdlog_logger& logger() except +
+
+
+logging_levels = [
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏I see this is being used in the tests. Do we want to expose this in the top-level namespace? I think it is fine, but if so:

<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Let us document this variable for the user, and (separately) make it clear that the order matters (since the binding layer is using the integer location to map to the relevant spdlog enum)

There’s no way of attaching a doc string to a variable such that the interpreter knows about it, but there is a sphinx convention. Either, follow the variable by a docstring, or prefix it with a comment using `#:`

```
#: docstring for a variable
#: defined below
some_var = ...

other_var = ...
"""docstring for variable above"""

Reply:
The Enum `logging_level` has a docstring.

Reply:
This is gone.